src/hb-ot-layout-gsubgpos.hh

```diff
@@ -2002,6 +2002,77 @@ struct ChainRule
 					      lookup.arrayZ, lookup_context));
   }
 
+  void serialize (hb_serialize_context_t *c,
+                  const hb_map_t *backtrack_map,
+                  const hb_map_t *input_map = nullptr,
+                  const hb_map_t *lookahead_map = nullptr) const
+  {
```

Comment:
Looking at other methods in this class, looks like you need something like this:

```
const HeadlessArrayOf<HBUINT16> &input = StructAfter<HeadlessArrayOf<HBUINT16>> (backtrack);
const ArrayOf<HBUINT16> &lookahead = StructAfter<ArrayOf<HBUINT16>> (input);
const ArrayOf<LookupRecord> &lookup = StructAfter<ArrayOf<LookupRecord>> (lookahead);
```

Too access the array members, since the members on this struct don't actually point to the real locations.